test/sqlite/sqllogic_test_runner.cpp

```diff
@@ -23,13 +23,55 @@ SQLLogicTestRunner::SQLLogicTestRunner(string dbpath) : dbpath(std::move(dbpath)
 	config->options.allow_unredacted_secrets = true;
 	config->options.load_extensions = false;
 
+	auto &test_config = TestConfiguration::Get();
+	loading_mode = test_config.GetExtensionLoadingMode();
+	install_mode = test_config.GetExtensionInstallMode();
+
+	config->options.autoload_known_extensions = false;
+	config->options.autoinstall_known_extensions = false;
+	config->options.allow_unsigned_extensions = true;
+	local_extension_repo = "";
+
 	auto env_var = std::getenv("LOCAL_EXTENSION_REPO");
-	if (!env_var) {
-		config->options.autoload_known_extensions = false;
-	} else {
+	if (env_var) {
 		local_extension_repo = env_var;
 		config->options.autoload_known_extensions = true;
 	}
+
+	switch (loading_mode) {
+	case TestConfiguration::ExtensionLoadingMode::NONE: {
+		break;
+	}
+	case TestConfiguration::ExtensionLoadingMode::AUTOLOAD_ONLY:
+	case TestConfiguration::ExtensionLoadingMode::ALL: {
+		config->options.autoload_known_extensions = true;
+		break;
+	}
+	}
+	if (config->options.autoinstall_known_extensions) {
+		switch (install_mode) {
+		case TestConfiguration::ExtensionInstallMode::NONE: {
+			break;
+		}
+		case TestConfiguration::ExtensionInstallMode::LOCAL_ONLY:
+		case TestConfiguration::ExtensionInstallMode::EITHER: {
+			config->options.autoinstall_known_extensions = true;
+			if (local_extension_repo.empty()) {
+				local_extension_repo = "build/release/repository";
+			}
+			break;
+		}
+		case TestConfiguration::ExtensionInstallMode::REMOTE_NO_CHECKS:
+		case TestConfiguration::ExtensionInstallMode::REMOTE_ONLY: {
+			config->options.allow_unsigned_extensions = false;
```

Comment:
Maybe this should be again another config option - `allow_unsigned` which defaults to true